packages/core/src/lib/toast/context.ts

```diff
@@ -0,0 +1,141 @@
+import { setContext, getContext } from 'svelte';
+import {
+  derived,
+  get,
+  writable,
+  type Readable,
+  type Writable,
+} from 'svelte/store';
+
+import { uniqueId } from '$lib/unique-id';
+import { ToastVariant, type ToastVariantType } from '$lib/toast';
+
+import { pausableProgress } from '../notification/pausable-progress';
+
+/** Internal toast context. */
+export interface ToastContext {
+  state: ToastState;
+  toast: Toast;
+}
+
+/** Notification sender. */
+export interface Toast {
+  /** Push a neutral toast with a message. */
+  neutral: (message: string, closeable?: boolean) => void;
+  /** Push an upload toast with a message. */
+  upload: (message: string, closeable?: boolean) => void;
+  /** Push a success toast with a message. */
+  success: (message: string, closeable?: boolean) => void;
+}
+
+/** Internal toast state. */
+export interface ToastState {
+  toasts: Readable<ToastElement[]>;
+  pageIsVisible: Writable<boolean>;
+}
+
+/** Internal toast item state and actions. */
+export interface ToastElement {
+  id: string;
+  message: string;
+  closeable: boolean | undefined;
+  variant: ToastVariantType;
+  progress: Readable<number>;
+  pause: () => void;
+  resume: () => void;
+  dismiss: () => void;
+}
+
+const ToastContextKey = Symbol('toast-context');
+const ToastDuration = 4000;
+
+/** Create the internal toast state and context object. */
+export const createToastContext = (): ToastContext => {
+  const toasts = writable<ToastElement[]>([]);
+  const pageIsVisible = writable(true);
+  const pausedID = writable<string | undefined>();
+  const unsubscribeByID = new Map<string, () => void>();
```

Comment:
Not an issue, but why a `Map` over a `Record`?

Reply:
Since we're dynamically adding and removing these handlers, I think a `Map` is better than a plain-object record. We have lint rules that will yell at you for using `delete obj[key]`